inc/user.class.php

```diff
@@ -4504,38 +4504,60 @@ static function getUniquePersonalToken() {
     * @param $ID user ID
     * @param string $field the field storing the token
     *
+    * @deprecated 9.2 @see User::getAuthToken()
+    *
     * @return string token
    **/
    static function getToken($ID, $field = 'personal_token') {
       global $DB;
 
+      Toolbox::deprecated('getToken() method is deprecated');
       $user = new self();
       if ($user->getFromDB($ID)) {
-         if (!empty($user->fields[$field])) {
-            return $user->fields[$field];
-         }
-         $token = self::getUniqueToken($field);
-         $user->update(['id'             => $user->getID(),
-                             $field           => $token,
-                             $field . "_date" => $_SESSION['glpi_currenttime']]);
-         return $user->fields[$field];
+         return $user->getAuthToken($field);
       }
 
       return false;
    }
 
+   /**
+    * Get token of a user. If it does not exists  then generate it.
+    *
+    * @param string $field the field storing the token
+    *
+    * @return string|false token or false in case of error
+    */
+   public function getAuthToken($field = 'personal_token') {
+      global $DB;
+
+      if ($this->isNewItem()) {
+         return false;
+      }
+
+      if (!empty($this->fields[$field])) {
+         return $this->fields[$field];
+      }
+      $token = self::getUniqueToken($field);
+      $this->update(['id'             => $this->getID(),
+                     $field           => $token,
+                     $field . "_date" => $_SESSION['glpi_currenttime']]);
+      return $this->fields[$field];
+   }
+
    /**
     * Get personal token of a user. If not exists generate it.
     *
     * @param $ID user ID
-   *
-   * @deprecated 9.2 @see User::getToken()
+    *
+    * @deprecated 9.2 @see User::getAuthToken()
     *
     * @return string personal token
-   **/
+    */
    static function getPersonalToken($ID) {
       Toolbox::deprecated('getPersonalToken() method is deprecated');
-      return self::getToken($ID, 'personal_token');
+      $user = new self();
+      $user->getFromDB($ID);
+      return $user->getAuthToken($ID, 'personal_token');
```

Comment:
The `$ID` param has to be removed and the `personal_token` values is default and could be removed too.